src/command_modules/azure-cli-extension/azure/cli/command_modules/extension/custom.py

```diff
@@ -259,3 +265,28 @@ def update_extension(extension_name, index_url=None, pip_extra_index_urls=None,
 
 def list_available_extensions(index_url=None):
     return get_index_extensions(index_url=index_url)
+
+def check_distro_consistency():
+    if not IS_WINDOWS:
+        try:
+            logger.debug('Linux distro check: Reading from: %s', LIST_FILE_PATH)
+
+            with open(LIST_FILE_PATH, 'r') as list_file:
+                package_source = list_file.read() 
+
+            stored_linux_dist_name = package_source.split(" ")[3]
+            logger.debug('Linux distro check: Found in list file: %s', stored_linux_dist_name)
+
+            current_linux_dist_name = platform.linux_distribution()[2]
+            logger.debug('Linux distro check: Reported by API: %s', current_linux_dist_name)
```

Comment:
Put the above 4 lines indented under the above `with` statement.
`package_source` would be defined in the scope of the `with` statement so the other code should be under it.